### PR TITLE
Symfony 5 and nexylan/slack 3.0 support

### PR DESCRIPTION
**Pull request by mbabker**
Also adds PHP 7.3 to the CI configuration

### CODE REVIEW

**Comment by mbabker** [comment]
Well, this ended up becoming more involved than I expected.

Basically, to get around dealing with differing signatures in different versions of `matthiasnoback/symfony-dependency-injection-test`, I ended up:

- Raising PHP minimum to 7.2
- Bumping PHPUnit to 8.x
- Bumping `matthiasnoback/symfony-dependency-injection-test` to 4.x

Only the first one in that list will affect bundle users, but considering that version is hours away from EOL at this point I'd say not an issue.

File: tests/NexySlackBundleTest.php [hunk 1]
```diff
@@ -33,6 +33,9 @@ protected function setUp(): void
         $this->bundle = new NexySlackBundle();
     }
 
+    /**
+     * @doesNotPerformAssertions
```
Comment (mbabker) [inline tests/NexySlackBundleTest.php]: PHPUnit marks tests that do not perform any assertions as risky, this annotation silences that.

File: phpunit.xml.dist [hunk 2]
```diff
@@ -8,8 +8,7 @@
          convertNoticesToExceptions="true"
          convertWarningsToExceptions="true"
          processIsolation="false"
-         stopOnFailure="false"
-         syntaxCheck="false">
```
Comment (soullivaneuh) [inline phpunit.xml.dist]: Why this removal?

File: tests/NexySlackBundleTest.php (hunk 1 above)
Comment (soullivaneuh) [reply tests/NexySlackBundleTest.php]: Nice catch! :+1:

**Comment by mbabker** [comment]
> Travis does not seem happy with your new Symfony v5 environment, would you mind to take a look?

It looks like this is blocked by https://github.com/php-http/HttplugBundle/pull/360, once HttplugBundle updated and has a new release in theory this should be OK.

File: phpunit.xml.dist (hunk 2 above)
Comment (mbabker) [reply phpunit.xml.dist]: PHPUnit 7 removed this attribute.

**Comment by mbabker** [comment]
HttplugBundle finally had a release yesterday.  So, bumping this bundle's minimum version of that up to the new version got most things passing.  But, the Symfony 5 build still couldn't resolve a full set of dependencies because the `nexylan/slack` package doesn't support Symfony 5 until the 3.0 release.  So, I've bumped that across major versions too and adjusted the Client class configuration to match the needed constructor changes.  That also in effect causes this bundle to require PHP 7.3 too.  Things should be good to go here now.

File: src/DependencyInjection/Configuration.php [hunk 3]
```diff
@@ -38,6 +38,8 @@ public function getConfigTreeBuilder(): TreeBuilder
                     ->addDefaultsIfNotSet()
                     ->children()
                         ->scalarNode('client')->defaultValue('httplug.client')->end()
+                        ->scalarNode('request_factory')->defaultValue('nexy_slack.request_factory.default')->end()
+                        ->scalarNode('stream_factory')->defaultValue('nexy_slack.stream_factory.default')->end()
```
Comment (mbabker) [inline src/DependencyInjection/Configuration.php]: Same logic as with the HTTP client class, allow a bundle user to replace the request and stream factories with their own implementation.

File: src/Resources/config/client.xml [hunk 4]
```diff
@@ -6,11 +6,21 @@
...
+            <argument type="service" id="nexy_slack.http.request_factory" />
+            <argument type="service" id="nexy_slack.http.stream_factory" />
             <argument>%nexy_slack.endpoint%</argument>
             <argument>%nexy_slack.config%</argument>
-            <argument type="service" id="nexy_slack.http.client" />
         </service>
 
         <service id="nexy_slack.client" alias="Nexy\Slack\Client"/>
+
+        <service id="nexy_slack.request_factory.default" class="Psr\Http\Message\RequestFactoryInterface">
```
Comment (mbabker) [inline src/Resources/config/client.xml]: HttplugBundle doesn't provide services directly creating the PSR-17 factories, so the services here are the defaults used to discover compatible factories through the Httplug discovery system.

**Comment by soullivaneuh** [comment]
Thank for your work and your time! :+1:

**Pull request merged**